src/Datadog.Trace/Agent/Api.cs

```diff
@@ -173,7 +171,7 @@ private static IApiRequestFactory CreateRequestFactory()
 #endif
         }
 
-        private async Task<bool> SendTracesAsync(Span[][] traces, IApiRequest request)
+        private async Task<bool> SendTracesAsync(Span[][] traces, IApiRequest request, bool finalTry)
```

Comment:
Could we instead pass in a callback to run when the status code is not 200 or 300? That way this method doesn't even have to know the difference between final try or not

Reply:
Sure, I didn't like this strategy either!

Reply:
Upon discussing that this would require passing an async lambda, we've decided to stick with passing a bool.